pkg/cluster/cluster.go

```diff
@@ -48,7 +48,14 @@ func (b *TemporalClusterBuilder) ResourceBuilders() ([]resource.Builder, error)
 			return nil, err
 		}
 		builders = append(builders, resource.NewServiceAccountBuilder(serviceName, b.Instance, b.Scheme, specs))
-		builders = append(builders, resource.NewDeploymentBuilder(serviceName, b.Instance, b.Scheme, specs))
+
+		// worker does not expose grpc endpoint so need to remove liveness probes
+		if serviceName == "worker" {
+			builders = append(builders, resource.NewWorkerDeploymentBuilder(serviceName, b.Instance, b.Scheme, specs))
```

Comment:
Instead of duplicating the whole builder, can't we add the condition in the deployment builder itself ? 
For instance:
```go
livenessProbe := nil
if serviceName != common.WorkerServiceName {
  livenessProbe := &corev1.Probe{ [...] }
}
```

What do you think about that ?

Reply:
yeah originally i was going that route but then i thought there might be more differences and it would be messy eventually to have logic in builder but sure im fine with that.